### PR TITLE
fix shellcheck in hack/make-rules/...

### PR DESCRIPTION
**Pull request by xichengliudui**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
fix shellcheck in hack/make-rules/...

**Which issue(s) this PR fixes**:
Ref #72956

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: hack/make-rules/test-cmd.sh
Comment (BenTheElder) [inline hack/make-rules/test-cmd.sh]: what are these echos about?

File: hack/make-rules/test-cmd.sh
Comment (BenTheElder) [inline hack/make-rules/test-cmd.sh]: why is this echoed?

File: hack/make-rules/test-cmd.sh
Comment (BenTheElder) [inline hack/make-rules/test-cmd.sh]: why is this echoed?

File: hack/make-rules/test-integration.sh
Comment (BenTheElder) [inline hack/make-rules/test-integration.sh]: same here

Comment (BenTheElder) [reply hack/make-rules/test-integration.sh]: cc @ixdy any idea why we have `KUBE_RACE=""` below?

**Review by BenTheElder — COMMENTED** [review]
the added `echo`s look like bypassing a lint without fixing the root cause

File: hack/make-rules/test-cmd.sh
Comment (xichengliudui) [reply hack/make-rules/test-cmd.sh]: SUPPORTED_RESOURCES is not used, so we have to delete or export

Comment (xichengliudui) [reply hack/make-rules/test-cmd.sh]: can we delete these unreferenced ones? If you don't choose, shellcheck looks like this:
```
SC2034: APISERVER_PID appears unused. Verify use (or export if used externally).
For more information: https://www.shellcheck.net/wiki/SC2034
```

Comment (BenTheElder) [reply hack/make-rules/test-cmd.sh]: possibly we should export it instead, there are some functions in other files that read this variable. 
https://cs.k8s.io/?q=SUPPORTED_RESOURCES&i=nope&files=&repos=kubernetes/kubernetes

Comment (BenTheElder) [reply hack/make-rules/test-cmd.sh]: this is one of those tricky things shellcheck can't detect, the only way to know is to trace what gets executed in everypath and see if something does read this.

File: hack/make-rules/make-help.sh
Comment (spiffxp) [inline hack/make-rules/make-help.sh]: this isn't an equivalent substitution

Comment (xichengliudui) [reply hack/make-rules/make-help.sh]: I will fix this file later.

**Comment by liggitt** [comment]
/unassign
feel free to add me back once this has lgtm

File: hack/make-rules/test-cmd.sh
Comment (spiffxp) [inline hack/make-rules/test-cmd.sh]: OK, so this is really difficult for me to review, and as a maintainer it looks like it introduces more complexity.  Help me out by answering the following: Why are you exporting this? What uses it? What are you fixing by doing this?

As a maintainer, if this export is necessary, this function should have a comment explaining at least what it exports

File: hack/make-rules/test-cmd.sh
Comment (spiffxp) [inline hack/make-rules/test-cmd.sh]: Same here

File: hack/make-rules/test-cmd.sh
Comment (spiffxp) [inline hack/make-rules/test-cmd.sh]: And here

File: hack/make-rules/test-integration.sh
Comment (spiffxp) [inline hack/make-rules/test-integration.sh]: And here

File: hack/make-rules/test-integration.sh
Comment (spiffxp) [inline hack/make-rules/test-integration.sh]: Why are you doing this? What shellcheck failure are you fixing by doing this?

**Review by spiffxp — DISMISSED** [review]
/hold
These changes are difficult to review because they lack context.  I feel like they are more driving to the letter than the spirit of fixing shellcheck failures.  We should be making our bash more maintainable and explaining how/why in the process.

File: hack/make-rules/test-cmd.sh
Comment (xichengliudui) [reply hack/make-rules/test-cmd.sh]: I did not express clearly is my problem, let me answer:
All the variables here and below are unused. using the shellcheck check looks like this:
```
SC2034: SUPPORTED_RESOURCES appears unused. Verify use (or export if used externally).
For more information: https://www.shellcheck.net/wiki/SC2034
```
Because some files might need them, I chose to export these variables. 
https://cs.k8s.io/?q=SUPPORTED_RESOURCES&i=nope&files=&repos=kubernetes/kubernetes
Please feel free to ask me if you have any questions.

File: hack/make-rules/test-integration.sh
Comment (xichengliudui) [reply hack/make-rules/test-integration.sh]: Why did I modify and shellcheck like this lead to what failure：
```
In ./hack/make-rules/test-integration.sh line 98:
IFS=';' read -a apiVersions <<< "${KUBE_TEST_API_VERSIONS}"
        ^--^ SC2162: read without -r will mangle backslashes.

For more information:
  https://www.shellcheck.net/wiki/SC2162 -- read without -r will mangle backs...
```

Comment (BenTheElder) [reply hack/make-rules/test-integration.sh]: should be `read -a` -> `read -ar` then, a is a different flag. https://ss64.com/bash/read.html

**Comment by xichengliudui** [comment]
If I don't delete `./hack/make-rules/test-integration.sh` I can resolve the conflict. But shellcheck fails
```
... skipping 36450 lines ...
Congratulations! All shell files are passing lint (excluding those in hack/.shellcheck_failures).
Some packages in hack/.shellcheck_failures are passing shellcheck. Please remove them.

  ./hack/make-rules/test-integration.sh
```
If I delete the `. / hack/make rules/test - integration. Sh ` conflict will occur and what is this, I wonder

**Comment by BenTheElder** [comment]
@xichengliudui please see: https://github.com/kubernetes/community/blob/a095f16dc8de303eb8ae104a231b8ab85a5a76d9/contributors/guide/github-workflow.md

`git fetch upstream` and `git rebase upstream/master` is the way to go sync, once you've synced to upstream with rebase you should be able to have the conflict fixed _and_ have that file removed from `hack/.shellcheck_failures`.

File: hack/make-rules/test-cmd.sh
Comment (spiffxp) [inline hack/make-rules/test-cmd.sh]: Sorry if I'm not being clear, I'm asking for you to add these comments above the function, not above the export line, eg:
```
# Runs kube-apiserver
#
# Exports:
#   APISERVER_PID
function run_kube_apiserver() {
```

File: hack/make-rules/test-cmd.sh
Comment (spiffxp) [inline hack/make-rules/test-cmd.sh]: Same here, for `run_kube_controller_manager`

File: hack/make-rules/test-cmd.sh
Comment (spiffxp) [inline hack/make-rules/test-cmd.sh]: Same here, for `create_node`

**Review by spiffxp — COMMENTED** [review]
almost there

File: hack/make-rules/test-cmd.sh
Comment (xichengliudui) [reply hack/make-rules/test-cmd.sh]: nice!

File: hack/make-rules/test-cmd.sh
Comment (BenTheElder) [reply hack/make-rules/test-cmd.sh]: comment was added, this may? actually be read

Comment (BenTheElder) [reply hack/make-rules/test-cmd.sh]: https://cs.k8s.io/?q=CTLRMGR_PID&i=nope&files=&repos=kubernetes/kubernetes

File: hack/make-rules/test-cmd.sh
Comment (BenTheElder) [reply hack/make-rules/test-cmd.sh]: comment was added. similarly this may(?) be read https://cs.k8s.io/?q=SUPPORTED_RESOURCES&i=nope&files=&repos=kubernetes/kubernetes

**Review by BenTheElder — APPROVED** [review]
/lgtm

**Comment by xichengliudui** [comment]
@sttts

**Comment by xichengliudui** [comment]
@BenTheElder @spiffxp , do we still need to merge this pr? I'll close it if I don't need to merge it.

**Comment by spiffxp** [comment]
I've been away from reviews for a while due to v1.14 release duties, sorry for the delay, thanks for your patience

**Comment by xichengliudui** [comment]
@spiffxp  Obviously the release of v1.14 is more important, thank you! I think https://github.com/kubernetes/kubernetes/pull/74481 needs you, too

**Pull request merged**
